### DISCUSSION TITLE
`FileNotFoundException` with NuGet dependencies

### DISCUSSION THREAD

**Discussion by dagillespie** [nxrighthere/UnrealCLR#108] [Help] [answered]
Hi,
This may have been answered tangentially elsewhere however I'm getting the following which suggests that UnrealCLR can't find NewtonSoft when referenced in Visual Studio.

- NewtonSoft is referenced in via Nuget package manager
- Project builds fine with this reference added and runs in UnrealCLR however if any code uses any NetwonSoft functionality, the error comes up.
- There doesn't appear to be any Newtonsoft dll's copied over and I would have expected that to already have been handled by Visual Studio.

`LogUnrealCLR: Error: UnrealCLR::Module::Exception: System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The system cannot find the file specified.
File name: 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
   at Project.ProjectClient.Client..ctor()
   at Project.ProjectClient.Main.OnWorldPostBegin()
   at UnrealEngine.Runtime.Core.ManagedCommand(Command command) in C:\REPO\UnrealCLR\Source\Managed\Runtime\Runtime.cs:line 150
LogUnrealCLR: Error: UnrealCLR::Module::Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at Project.ProjectClient.Main.OnWorldPostPhysicsTick(Single deltaTime)
   at UnrealEngine.Runtime.Core.ManagedCommand(Command command) in C:\REPO\UnrealCLR\Source\Managed\Runtime\Runtime.cs:line 155
LogUnrealCLR: Error: UnrealCLR::Module::Exception: System.NullReferenceException: Object reference not set to an instance of an object.
etc....`

**Comment by nxrighthere** [comment, marked as answer]
You need to publish your project, not just build it.

**Reply by dagillespie** [reply]
Thanks - just tried publishing as a both "self contained" and "framework independent" which both failed... however this led to the following which was overlooked in the .csproj

`<CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>`

With this present in built or published projects the dependencies appear to be copied and referenced correctly.

https://stackoverflow.com/questions/45464271/include-nuget-dependencies-in-my-build-output
